### PR TITLE
Prac 05 feedback

### PR DESCRIPTION
**Pull request by Dajin-01**
@Winminoo4444

### CODE REVIEW

File: prac_05/state_names.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""
+CP1404/CP5632 Practical
+State names in a dictionary
+File needs reformatting
+"""
+
+# TODO: Reformat this file so the dictionary code follows PEP 8 convention
+CODE_TO_NAME = {
```
Comment (Winminoo4444) [inline prac_05/state_names.py:8]: You follow the guidelines for the subject dictionaries "key_to_value". Good Job!

File: prac_05/hex_colours.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "beige": "#f5f5dc",
+    "bisque": "#ffe4c4",
+    "black": "#000000",
+    "BlanchedAlmond": "#ffebcd",
+    "blue": "#0000ff"
+}
+
+
+def main():
+    """Get color names until enter nothing."""
```
Comment (Winminoo4444) [inline prac_05/hex_colours.py:22]: """Get color names from the user and display their hex codes.""" might be a better comment here. Yours is fine too.

File: prac_05/word_occurrences.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+def main():
+    # sorts the words alphabetically, and prints each word with its count.
+    text = input("Text: ")
```
Comment (Winminoo4444) [inline prac_05/word_occurrences.py:3]: The current implementation is case-sensitive, meaning that "Hello" and "hello" will be treated as different words. If you want to ignore case, you can convert all words to lowercase as optional. Like "text = input("Text: ").lower()".

File: prac_05/wimbledon.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    with open(filename, "r", encoding="utf-8-sig") as in_file:
+        next(in_file)
+        for line in in_file:
+            data = line.strip().split(',')
+            year = data[0]
+            country = data[1]
+            champion = data[2]
+            champions_data.append((year, country, champion))
+    return champions_data
+
```
Comment (Winminoo4444) [inline prac_05/wimbledon.py:13]: In the case if the file does not exist or is unreadable for some reason, the program will raise an unhandled exception. You can improve this by adding error handling.

File: prac_05/wimbledon.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    for _, _, champion in champions_data:
+        if champion in champion_count:
+            champion_count[champion] += 1
+        else:
+            champion_count[champion] = 1
+    return champion_count
+
+
+def get_countries(champions_data):
+    """Get a sorted set of countries of the champions."""
```
Comment (Winminoo4444) [inline prac_05/wimbledon.py:27]: Since the function returns a sorted list, you can name the function "get_sorted_countries".

**Review by Winminoo4444 — COMMENTED** [review]
Good attempt for this practical, my friend. I have learned a lot while reviewing your code.

**Comment by Dajin-01** [comment]
Thank you for your feedback, it has helped me a lot with my programming skills

File: prac_05/wimbledon.py (hunk 5 above)
Comment (Dajin-01) [reply prac_05/wimbledon.py:27]: Thank you for your feedback, it has helped me a lot with my programming skills

**Pull request merged**